modules/apps/29-fee/types/msgs.go

```diff
@@ -41,3 +44,135 @@ func (msg MsgRegisterCounterpartyAddress) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{signer}
 }
+
+// NewMsgPayPacketFee creates a new instance of MsgPayPacketFee
+func NewMsgPayPacketFee(fee Fee, sourcePortId, sourceChannelId, signer string, relayers []string) *MsgPayPacketFee {
+	return &MsgPayPacketFee{
+		Fee:             fee,
+		SourcePortId:    sourcePortId,
+		SourceChannelId: sourceChannelId,
+		Signer:          signer,
+		Relayers:        relayers,
+	}
+}
+
+// ValidateBasic performs a basic check of the MsgPayPacketFee fields
+func (msg MsgPayPacketFee) ValidateBasic() error {
+	// validate channelId
+	err := host.ChannelIdentifierValidator(msg.SourceChannelId)
+	if err != nil {
+		return err
+	}
```

Comment:
nit/suggestion (and applied when necessary):
```suggestion
	if err := host.ChannelIdentifierValidator(msg.SourceChannelId); err != nil {
		return err
	}
```